### ISSUE TITLE
Cannot export complex arrays as dlpack

### ISSUE THREAD

**Issue by izaid** [google/jax#9497]
To my knowledge, dlpack now supports complex numbers (see https://github.com/dmlc/dlpack/blob/main/include/dlpack/dlpack.h#L125), but Jax throws an error when trying to turn a complex array to a dlpack.

This bug, combined with the other one I just filed - https://github.com/google/jax/issues/9496 - actually creates a blocking issue of being unable to share complex number arrays from Jax with other array packages (except for NumPy, where it works due to Jax supporting `__array_interface__`). I'd be keen to know if there is any possible workaround for this. How can I share a complex array with CuPy, for example?

```python
import numpy as np

import jax
import jax.dlpack
import jax.numpy as jnp

a = jnp.zeros(10, dtype = jnp.complex64)
jax.dlpack.to_dlpack(a)
# RuntimeError: UNIMPLEMENTED: XLA type C64 has no DLPack equivalent
```

**Comment by jakevdp** [comment]
Thanks for the report – if complex types are available in the dlpack version that jaxlib is built against, we'll have to update https://github.com/tensorflow/tensorflow/blob/5968de3893ebc49a4f2a0f3b0b498a8b20c6f145/tensorflow/compiler/xla/python/dlpack.cc#L98-L103 and https://github.com/google/jax/blob/d262bae88b53365df222242bf1f4a5b7498c8210/jax/_src/dlpack.py#L21-L23

**Comment by PhilipVinc** [comment]
0.4 IIRC

**Comment by izaid** [comment]
Yeah, looks like 0.4, https://github.com/dmlc/dlpack/commit/56e6b6c96597074d718b770e65ab8eea99ec2133

Hopefully this is an easy change ...

**Comment by hawkinsp** [comment]
Yes, this turns out to be a fairly easy fix. It should make it into the next jaxlib.

**Comment by izaid** [comment]
@hawkinsp That's great news, glad it was so easy. Many thanks!

**Comment by hawkinsp** [comment]
We just released `jax` and `jaxlib` 0.3.0, which should have this fix (lightly tested...). Please try it out!

**Comment by izaid** [comment]
@hawkinsp Many thanks, confirming that everything seems good to me - can export and import complex arrays on the GPU now!
